### PR TITLE
feat: add a debug mode for bentoctl

### PR DESCRIPTION
**Pull request by jjmachan**
#### Description

closes:

### CODE REVIEW

File: bentoctl/cli/__init__.py [hunk 1]
```diff
@@ -147,7 +146,6 @@ def build(
     generate_deployable_container(
         tag=local_docker_tag,
         deployment_config=deployment_config,
-        cleanup=get_debug_mode(),
```
Comment (ssheng) [inline bentoctl/cli/__init__.py:150]: Shouldn't we clean up when not in debug mode? :)

File: bentoctl/docker_utils.py [hunk 2]
```diff
@@ -51,9 +52,13 @@ def __rich_console__(self, *_):
...
 def generate_deployable_container(
-    tag: str, deployment_config: DeploymentConfig, cleanup: bool
+    tag: str, deployment_config: DeploymentConfig
 ) -> None:
-    with TempDirectory(cleanup=cleanup) as dist_dir:
+    """
+    Calls the operator and generates the deployable. If in debug more the generated
+    deployable will be moved to the current directory for easier debugging.
+    """
+    with TempDirectory(debug=get_debug_mode()) as dist_dir:
```
Comment (ssheng) [inline bentoctl/docker_utils.py:61]: Moving directory is a bit unnatural for debugging. Maybe preserving and printing the temp directory during debug should be sufficient?

File: bentoctl/utils/temp_dir.py [hunk 3]
```diff
@@ -46,4 +50,10 @@ def cleanup(self, ignore_errors=False):
         """
         if self.path is not None and os.path.exists(self.path):
             shutil.rmtree(self.path, ignore_errors=ignore_errors)
```
Comment (ssheng) [inline bentoctl/utils/temp_dir.py:52]: If in debug mode, maybe we can print the temp dir instead of removing.

**Comment by jjmachan** [comment]
closing this since we felt it would be better to print out the path for the intermediate bento instead of moving it to current path. #182 addresses the new change

**Pull request closed without merging**